arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1681,8 +1681,9 @@ Result ClusterInfo::createCollectionsCoordinator(std::string const& databaseName
   auto cacheMutexOwner = std::make_shared<std::atomic<std::thread::id>>();
   auto isCleaned = std::make_shared<bool>(false);
```

Comment:
Is there any reason why those are not in a single struct but instead we are using 6 (!) shared pointers?

Reply:
Good point, but let's address this separately in devel, and not in this branch (which should go to 3.5 as soon as it is reviewed & tested).